tests/phpunit/tests/theme/globalStylesheet.php

```diff
@@ -0,0 +1,79 @@
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--x-large: 42px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--custom: 100px;" ) );
+
+        $styles = wp_get_global_stylesheet( array( 'presets' ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--small: 13px" ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--medium: 20px" ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--large: 36px" ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--x-large: 42px" ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--custom: 100px;" ) );
+
+        $styles = wp_get_global_stylesheet();
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--small: 13px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--medium: 20px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--large: 36px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--x-large: 42px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--custom: 100px;" ) );
+    }
+
+    public function test_variables_in_classic_theme_with_no_presets() {
+        switch_theme( 'default' );
+
+        $styles = wp_get_global_stylesheet( array( 'variables' ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--small: 13px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--medium: 20px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--large: 36px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--x-large: 42px" ) );
+
+        $styles = wp_get_global_stylesheet( array( 'presets' ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--small: 13px" ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--medium: 20px" ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--large: 36px" ) );
+        $this->assertFalse( str_contains( $styles, "--wp--preset--font-size--x-large: 42px" ) );
+
+        $styles = wp_get_global_stylesheet();
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--small: 13px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--medium: 20px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--large: 36px" ) );
+        $this->assertTrue( str_contains( $styles, "--wp--preset--font-size--x-large: 42px" ) );
+    }
+
+    public function test_variables_in_classic_theme_with_presets() {
+        switch_theme( 'classic-with-presets' );
+
+        $this->assertTrue( current_theme_supports( 'editor-font-sizes') );
```

Comment:
I wasn't able to load the theme supports for this test theme (`classic-with-presets`). They don't seem to be loaded at this point, although they are declared in its `function.php` file ([see](https://github.com/WordPress/wordpress-develop/pull/2233/files#diff-3a386ff22169436c19bbfe68208390693ce80ab3db20cd8d7f4fef3ac542e7faR3)).

Reply:
For some reason, some environments didn't pick up the supports from the `functions.php` of the new test theme I created for this. Investigated a bit but didn't get anywhere. Switched the mechanism to adding the theme support dynamically in https://github.com/WordPress/wordpress-develop/pull/2233/commits/a78a74e010a1ee7e87f0783131f2ae63679b535f